### PR TITLE
Upgrade to Camel Quarkus 2.3.0, make the examples runnable against alternative BOMs

### PR DESCRIPTION
**Pull request by ppalaga**
Fix https://github.com/apache/camel-quarkus/issues/3152

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by jamesnetherton — APPROVED** [review]

File: kafka/README.adoc [hunk 1]
```diff
@@ -9,9 +9,13 @@ and other general information.
...
 
-The example application requires a running Kafka instance.
+The example application requires a Kafka instance.
 
-For simplicity, this example uses Quarkus DevServices in dev mode. It will automatically start Strimzi container for your tests, and you don’t even need to configure anything: the container will be automatically wired to the Quarkus dev mode configuration.
+You do not need to provide the Kafka instance yourself
+as long as you play with the example code in dev mode (a.k.a. `mvn quarkus:dev` - read more [here](https://quarkus.io/guides/getting-started#development-mode))
+or as long as you only run the supplied tests (`mvn test`).
+In those situations, Quarkus tooling starts a Redpanda image for you via [Quarkus Dev Services](https://quarkus.io/guides/kafka-dev-services)
+and it also configures the application so that you do not need touch anything in `application.properties`.
```
Comment (ppalaga) [inline kafka/README.adoc:19]: I reworded this a little bit. @jamesnetherton or @aldettinger could you please review, whether it makes sense?

File: kafka/src/main/resources/application.properties [hunk 2]
```diff
@@ -23,8 +23,7 @@ timer.period = 10000
 timer.delay = 10000
 
 # Kafka instance without Authentication
-# Passing ${kafka.bootstrap.servers} is a workaround for https://github.com/apache/camel-quarkus/issues/2939
-camel.component.kafka.brokers=${brokers:${kafka.bootstrap.servers}}
+#camel.component.kafka.brokers=${brokers}
```
Comment (ppalaga) [inline kafka/src/main/resources/application.properties:26]: https://github.com/apache/camel-quarkus/issues/2939 not being an issue anymore, I hope commenting the line back and removing ${kafka.bootstrap.servers} makes up a valid suggestion for end users?

File: kafka/README.adoc (hunk 1 above)
Comment (jamesnetherton) [reply kafka/README.adoc:19]: Looks ok to me.

File: kafka/src/main/resources/application.properties (hunk 2 above)
Comment (jamesnetherton) [reply kafka/src/main/resources/application.properties:26]: I think it's safe. It's how the previous impl was configured.

**Comment by ppalaga** [comment]
Hm... the Kafka example fails in native mode. Isn't the dev service magic supposed to work there as well, @jamesnetherton ?

**Comment by jamesnetherton** [comment]
> Hm... the Kafka example fails in native mode. Isn't the dev service magic supposed to work there as well, @jamesnetherton ?

Yes, it should. Maybe restore the workaround for now and I can investigate further since I'm working in that area at the moment anyway.

**Comment by ppalaga** [comment]
> > Hm... the Kafka example fails in native mode. Isn't the dev service magic supposed to work there as well, @jamesnetherton ?
> 
> Yes, it should. Maybe restore the workaround for now and I can investigate further since I'm working in that area at the moment anyway.

`camel.component.kafka.brokers=${brokers:${kafka.bootstrap.servers}}` helps in native mode, but fails in JVM mode with a message saying something like `kafka.bootstrap.servers` is not defined. Let me file an issue and disable the native test.

**Pull request merged**